tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,365 @@
+        assert os.path.exists(os.path.join(workspace_path, "configuration.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "model.py"))
+        assert os.path.exists(os.path.join(workspace_path, "data.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "data_pipeline.py"))
+
+        # 2. Raising Error of building workspace with already created path
+        with pytest.raises(FileExistsError):
+            otx_builder.build_task_config(**inputs)
+
+        # 3. Update hparam.yaml with train_type="selfsl"
+        workspace_path = os.path.join(tmp_dir_path, "test_builder_build_task_config_2")
+        train_type = "selfsl"
+        inputs = {
+            "task_type": "classification",
+            "train_type": train_type,
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        otx_builder.build_task_config(**inputs)
+        assert os.path.exists(workspace_path)
+        assert os.path.exists(os.path.join(workspace_path, "configuration.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "data.yaml"))
+        template = MPAConfig.fromfile(os.path.join(workspace_path, "template.yaml"))
+        expected_template_train_type = {"default_value": "SELFSUPERVISED"}
+        assert template.hyper_parameters.parameter_overrides.algo_backend.train_type == expected_template_train_type
+        model_dir = os.path.join(workspace_path, train_type)
+        assert os.path.exists(model_dir)
+        assert os.path.exists(os.path.join(model_dir, "model.py"))
+        assert os.path.exists(os.path.join(model_dir, "data_pipeline.py"))
+
+        # 4. Raising ValueError with wrong train_type
+        workspace_path = os.path.join(tmp_dir_path, "test_builder_build_task_config_3")
+        train_type = "unexpected"
+        inputs = {
+            "task_type": "classification",
+            "train_type": train_type,
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        with pytest.raises(ValueError):
+            otx_builder.build_task_config(**inputs)
+
+        # 5. Build workspace with model_type argments
+        workspace_path = os.path.join(tmp_dir_path, "test_builder_build_task_config_4")
+        model_type = "yolox"
+        inputs = {
+            "task_type": "detection",
+            "model_type": model_type,
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        otx_builder.build_task_config(**inputs)
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        template = MPAConfig.fromfile(os.path.join(workspace_path, "template.yaml"))
+        assert template.name.lower() == model_type
+
+        # 6. Raise ValueError when build workspace with wrong model_type argments
```

Comment:
I think there are many test cases in a single test case. how about separating them as separate test case?

Reply:
This is a design choice. I first proposed a 1 test function design for 1 function. What you say makes sense. The design is to further break down the function. It seems that discussions between groups are necessary.

Reply:
Anyone please comment here!

Reply:
Actually, when I learned unit test in the book, I read that "One unit test case should test only one thing.".
Based on this, I think separation is better choice. but as you said, this is a design choice in a way. So, it's good to listen other opinions.

Reply:
I prefer @eunwoosh 's opinion. ("One test with one method call and one statement that asserts or verifies the behavior was as expected")

Reply:
I reflected the opinions of Eunwoo and Sungman. Please confirm. https://github.com/openvinotoolkit/training_extensions/pull/1519/commits/fc590a79fddc0d6c695a186f904762ae4aea5f92